### DISCUSSION TITLE
clang-tidy and sanitization

### DISCUSSION THREAD

**Discussion by ArsenArsen** [managarm/managarm#309] [Ideas and RFCs]
In light of a recent raise of activity in this forum, I'll document a long running thought I had here: It would likely be beneficial for us to run clang-tidy with various sets of rules across parts of the codebase. This allows us to enforce some style questions (identifier styles, for instance, such as naming private members etcetera), and eliminate some obvious bugs (especially c++ footguns) in our code. This would probably take some effort, as clang-tidy is quite extensive, with a lot of configuration flags and fine tuning to be done.

It is likely we will need to do something about introducing the [Guidelines Support Library](https://github.com/Microsoft/GSL), or implementing our own (which should not be very difficult) in case we proceed.

**Comment by 64** [comment]
https://github.com/mesonbuild/meson/issues/7662 is a problem for us
